cmd/data.js

```diff
@@ -56,25 +61,54 @@ class Data {
     if (link) {
       const parsed = parseLink(link)
       if (!parsed) throw ERR_INVALID_INPUT(`Link "${link}" is not a valid key`)
-      const result = await this.ipc.data({ resource: 'link', secrets, link })
+      const result = this.ipc.data({ resource: 'link', secrets, link })
       await output(json, result, { tag: 'link' }, this.ipc)
     } else {
-      const result = await this.ipc.data({ resource: 'apps', secrets })
+      const result = this.ipc.data({ resource: 'apps', secrets })
       await output(json, result, { tag: 'apps' }, this.ipc)
     }
   }
 
   async dht (cmd) {
     const { command } = cmd
     const { json } = command.parent.flags
-    const result = await this.ipc.data({ resource: 'dht' })
+    const result = this.ipc.data({ resource: 'dht' })
     await output(json, result, { tag: 'dht' }, this.ipc)
   }
 
   async gc (cmd) {
     const { command } = cmd
     const { json } = command.parent.flags
-    const result = await this.ipc.data({ resource: 'gc' })
+    const result = this.ipc.data({ resource: 'gc' })
     await output(json, result, { tag: 'gc' }, this.ipc)
   }
+
+  async reset (cmd) {
+    const { command } = cmd
+    const { yes } = command.flags
+
+    if (!yes) {
+      const dialog = `${ansi.warning} Clearing database ${ansi.bold(PLATFORM_HYPERDB)}\n\n`
+      const ask = 'Type DELETE to confirm'
+      const delim = '?'
+      const validation = (val) => val === 'DELETE'
+      const msg = '\n' + ansi.cross + ' uppercase DELETE to confirm\n'
+      await confirm(dialog, ask, delim, validation, msg)
+    }
+
+    const complete = await pick(this.ipc.dataReset(), (tag) => tag === 'complete')
+    complete ? status('Success\n', true) : status('Failure (ipc.dataReset)\n', false)
+  }
+}
+
+function pick (stream, predicate) {
```

Comment:
@davidmarkclements can `pick ()` be moved somewhere more general? doesn't feel like belongs in here. Where?